pybamm/geometry/geometry.py

```diff
@@ -30,10 +30,20 @@ def __init__(self, *geometries, custom_geometry={}):
                 geometry = Geometry3DMacro()
             elif geometry == "1+1D macro":
                 geometry = Geometry1p1DMacro()
+            elif geometry == "2+1D macro":
+                geometry = Geometry2p1DMacro()
             elif geometry == "1D micro":
                 geometry = Geometry1DMicro()
             elif geometry == "1+1D micro":
                 geometry = Geometry1p1DMicro()
+            elif geometry == "(1+0)+1D micro":
+                geometry = Geometry1p0p1DMicro()
+            elif geometry == "(1+1)+1D micro":
+                geometry = Geometry1p1p1DMicro()
+            elif geometry == "(2+0)+1D micro":
+                geometry = Geometry2p0p1DMicro()
+            elif geometry == "(2+1)+1D micro":
+                geometry = Geometry2p1p1DMicro()
```

Comment:
this now looks like the class `Geometry` should take 3 parameters (current collector dimensions (0,1,2), through-cell dimensions (0,1) and particle dimensions (0,1), and we probably don't need a different subclass for each combination